controllers/builder_test.go

```diff
@@ -223,13 +228,13 @@ service:
 				},
 				&corev1.ConfigMap{
 					ObjectMeta: metav1.ObjectMeta{
-						Name:      "test-collector",
+						Name:      "test-collector-" + goodConfigHash,
 						Namespace: "test",
 						Labels: map[string]string{
 							"app.kubernetes.io/component":  "opentelemetry-collector",
 							"app.kubernetes.io/instance":   "test.test",
 							"app.kubernetes.io/managed-by": "opentelemetry-operator",
-							"app.kubernetes.io/name":       "test-collector",
+							"app.kubernetes.io/name":       "test-collector-" + goodConfigHash,
```

Comment:
The name of the object changing is fine, but I think we should keep this app name consistent with the collector intended to mount it. I.e. the guarantee that `kubectl get all -l app.kubernetes.io/name=<col-name>` is one we should aim to keep

Reply:
fair point, i changed this